src/runtime_src/core/pcie/windows/alveo/device_windows.cpp

```diff
@@ -1187,6 +1197,86 @@ struct mailbox
   }
 }; //end of struct mailbox
 
+
+struct aim_counter
+{
+  using result_type = query::aim_counter::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
+  {
+    const auto dbg_ip_data = boost::any_cast<query::aim_counter::debug_ip_data_type>(arg1);
+
+    return xrt_core::debug_ip::get_aim_counter_result(device, dbg_ip_data);
+  }
+};
+
+struct am_counter
+{
+  using result_type = query::am_counter::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
+  {
+    const auto dbg_ip_data = boost::any_cast<query::am_counter::debug_ip_data_type>(arg1);
+
+    return xrt_core::debug_ip::get_am_counter_result(device, dbg_ip_data);
+  }
+};
+
+struct asm_counter
+{
+  using result_type = query::asm_counter::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
+  {
+    const auto dbg_ip_data = boost::any_cast<query::asm_counter::debug_ip_data_type>(arg1);
+
+    return xrt_core::debug_ip::get_asm_counter_result(device, dbg_ip_data);
+  }
+};
+
+struct lapc_status
+{
+  using result_type = query::lapc_status::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
+  {
+    const auto dbg_ip_data = boost::any_cast<query::lapc_status::debug_ip_data_type>(arg1);
+
+    return xrt_core::debug_ip::get_lapc_status(device, dbg_ip_data);
+  }
+};
+
+struct spc_status
+{
+  using result_type = query::spc_status::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
+  {
+    const auto dbg_ip_data = boost::any_cast<query::spc_status::debug_ip_data_type>(arg1);
+
+    return xrt_core::debug_ip::get_spc_status(device, dbg_ip_data);
+  }
+};
+
+struct accel_deadlock_status
+{
+  using result_type = query::accel_deadlock_status::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
```

Comment:
Here is it static and not virtual member function?
Why?
I guess there is a huge lack of comments at the top of the files to explain the design and what is going on.

Reply:
This is the same issue as addressed by Soren in the following comment.